### DISCUSSION TITLE
Limit computer resources for running ISCE

### DISCUSSION THREAD

**Discussion by fdelgadodelapuente** [isce-framework/isce2#716] [General]
Dear ISCE developers,

Is there a way to set a maximum limit for the RAM, threads and cores that ISCE uses to run topsApp and the TOPS stack processor?

Francisco

**Comment by scottstanie** [comment]
https://github.com/isce-framework/isce2/discussions/412 : Setting `OMP_NUM_THREADS` in the terminal before you run the scripts will limit the CPU usage.

I don't think there's as straightforward a way to put a hard limit on the RAM though.

**Comment by scottstanie** [comment]
Oh my mistake on linking to that issue, I didn't notice it was you who asked it haha

**Comment by EJFielding** [comment]
Hi Francisco, I don't know of any standard Linux or UNIX (macOS) operating system way to limit how much RAM a given job is allowed to use. Maybe some of the special job control systems for computing clusters have that. When a running job asks for the memory allocation it needs for the step it is doing, then the operating system (OS) will normally supply that amount unless it cannot. Then the OS tries to optimize its RAM usage by swapping out some parts of jobs to the swap storage if it is low on RAM. The swapping process is orders of magnitude slower than RAM access, so it causes the computer to become unresponsive.
